api/scpca_portal/test/test_loader.py

```diff
@@ -0,0 +1,311 @@
+        self.assertIn(
+            "SCPCP999990/SCPCS999990/SCPCL999990_filtered_rna.h5ad", library0.data_file_paths
+        )
+        self.assertIn("SCPCP999990/SCPCS999990/SCPCL999990_processed.rds", library0.data_file_paths)
+        self.assertIn(
+            "SCPCP999990/SCPCS999990/SCPCL999990_processed_rna.h5ad", library0.data_file_paths
+        )
+        self.assertIn("SCPCP999990/SCPCS999990/SCPCL999990_qc.html", library0.data_file_paths)
+        self.assertIn(
+            "SCPCP999990/SCPCS999990/SCPCL999990_unfiltered.rds", library0.data_file_paths
+        )
+        self.assertIn(
+            "SCPCP999990/SCPCS999990/SCPCL999990_unfiltered_rna.h5ad", library0.data_file_paths
+        )
+
+        self.assertIn(Library.FileFormats.SINGLE_CELL_EXPERIMENT, library0.formats)
+        self.assertIn(Library.FileFormats.ANN_DATA, library0.formats)
+        self.assertFalse(library0.has_cite_seq_data)
+        self.assertFalse(library0.is_multiplexed)
+        # Assert that metadata attribute has been populated and did not default to empty dict
+        self.assertFalse(library0.metadata == {})
+        self.assertEqual(library0.modality, Library.Modalities.SINGLE_CELL)
+        self.assertEqual(library0.workflow_version, "development")
+
+        # CHECK PROJECT SUMMARIES VALUES
+        self.assertEqual(project.summaries.count(), 4)
+        summaries = project.summaries.all()
+        # Refers to SCPCSXX90
+        summary0 = summaries[0]
+        self.assertEqual(summary0.diagnosis, "diagnosis1")
+        self.assertEqual(summary0.sample_count, 1)
+        self.assertEqual(summary0.seq_unit, "cell")
+        self.assertEqual(summary0.technology, "10Xv3")
+
+        # Refers to SCPCSXX91
+        summary1 = summaries[1]
+        self.assertEqual(summary1.diagnosis, "diagnosis2")
+        self.assertEqual(summary1.sample_count, 1)
+        self.assertEqual(summary1.seq_unit, "spot")
+        self.assertEqual(summary1.technology, "visium")
+
+        # summmary[2] is a reference to SCPCSXX94
+        # which is a bug because this sample has no metadata / data files
+
+        # Refers to SCPCSXX97
+        summary3 = summaries[3]
+        self.assertEqual(summary3.diagnosis, "diagnosis3")
+        self.assertEqual(summary3.sample_count, 1)
+        self.assertEqual(summary3.seq_unit, "cell")
+        self.assertEqual(summary3.technology, "10Xv3")
+
+        # CHECK CONTACTS
+        self.assertEqual(project.contacts.count(), 2)
+        contacts = project.contacts.all()
+        for contact_idx, contact in enumerate(contacts):
+            self.assertEqual(contact.name, "{contact " + str(contact_idx + 1) + "}")
+            self.assertEqual(contact.email, "{email contact " + str(contact_idx + 1) + "}")
+            self.assertEqual(contact.pi_name, "scpca")
+
+        # CHECK EXTERNAL ACCESSION VALUES
+        self.assertEqual(project.external_accessions.count(), 2)
+
+        ea0, ea1 = project.external_accessions.all()
```

Comment:
Query one a time by their accessions. Then assert on `has_raw` and `url` (`accession` will have been tested by the ability to grab the correct `external_accession`)